.github/workflows/test-go.yaml

```diff
@@ -27,20 +27,60 @@ jobs:
 
       - name: Check Go modules
         run: |
-          go mod tidy && pushd hack/swagger && go mod tidy && popd && git add go.* &&
+          go mod tidy &&
           git diff --cached --exit-code || (echo 'Please run "go mod tidy" to sync Go modules' && exit 1);
       - name: Check auto-generated assets
         run: |
-          make generate && git add pkg sdk manifests &&
+          make generate &&
           git diff --cached --exit-code || (echo 'Please run "make generate" to generate assets' && exit 1);
-      - name: Verify gofmt
+      - name: Check go fmt
         run: |
-          make fmt && git add pkg cmd &&
+          make fmt &&
           git diff --cached --exit-code || (echo 'Please run "make fmt" to verify gofmt' && exit 1);
-      - name: Verify govet
+      - name: Check go fmt
         run: |
-          make vet && git add pkg cmd &&
+          make vet
           git diff --cached --exit-code || (echo 'Please run "make vet" to verify govet' && exit 1);
-      - name: Verify golint
+      - name: Check golangci lint
         run: |
           make golangci-lint
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    env:
+      GOPATH: ${{ github.workspace }}/go
+    defaults:
+      run:
+        working-directory: ${{ env.GOPATH }}/src/github.com/kubeflow/training-operator
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # Kubernetes versions for setup-envtest integration tests.
+        kubernetes-version: ["1.29.3", "1.30.0", "1.31.0"]
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          path: ${{ env.GOPATH }}/src/github.com/kubeflow/training-operator
+
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: ${{ env.GOPATH }}/src/github.com/kubeflow/training-operator/go.mod
+
+      - name: Run Go unit tests
+        run: |
+          make test
+
+      - name: Run Go integration tests
+        run: |
+          make test-integration ENVTEST_K8S_VERSION=${{ matrix.kubernetes-version }}
+
+      - name: Coveralls report
```

Comment:
Our Coveralls report doesn't work, so we might need to re-visit this: https://coveralls.io/github/kubeflow/training-operator?branch=master